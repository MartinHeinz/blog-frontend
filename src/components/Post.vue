<template>
    <article class="post" itemscope="" itemtype="http://schema.org/BlogPosting">

        <div class="content" itemprop="articleBody">
            <p>{{ text }}</p>
        </div>
    </article>
</template>

<script>
import axios from 'axios';
import { API_URL } from '@/common/config';

export default {
    name: 'Post',
    props: {
        text: String,
    },
    watch: {
        $route: 'updateContent',
    },
    methods: {
        updateContent() {
            axios
                .get(`${API_URL}posts/${this.$route.params.id}`)
                .then((response) => {
                    this.text = response.data.text;
                });
        },
    },
};
</script>

<style>

    span {}

</style>
