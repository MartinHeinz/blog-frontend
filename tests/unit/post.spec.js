import { createLocalVue, shallowMount } from '@vue/test-utils';
import Post from '@/components/Post.vue';

import Vue from 'vue';
import Vuex from 'vuex';

const text = 'This is blog about something else...';
const $route = {
    hash: '',
    params: { id: '2' },
};

describe('Post.vue', () => {
    Vue.use(Vuex);
    let getters;
    let actions;
    let store;

    beforeEach(() => {
        getters = {
            currentPostText: () => ({ template: `<div>${text}</div>` }),
        };
        actions = {
            fetchPostById: jest.fn(),
        };

        store = new Vuex.Store({
            getters,
            actions,
        });
    });

    it('Sets text of post', () => {
        const localVue = createLocalVue();

        const wrapper = shallowMount(Post, {
            localVue,
            store,
            mocks: {
                $route,
            },
        });
        expect(wrapper.html()).toMatchSnapshot(); // TODO Fix <anonymous-stab>
    });
});
