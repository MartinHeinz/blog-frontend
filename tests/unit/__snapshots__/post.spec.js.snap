// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post.vue Sets text of post 1`] = `
<article itemscope="itemscope" itemtype="http://schema.org/BlogPosting" class="post">
  <div itemprop="articleBody" class="content">
    <anonymous-stub></anonymous-stub>
  </div>
</article>
`;
